pkg/controller/linstorcontrollerset/linstorcontrollerset_controller.go

```diff
@@ -191,10 +191,6 @@ func (r *ReconcileLinstorControllerSet) Reconcile(request reconcile.Request) (re
 		"spec":      fmt.Sprintf("%+v", pcs.Spec),
 	}).Debug("found LinstorControllerSet")
 
-	if pcs.Status.SatelliteStatuses == nil {
-		pcs.Status.SatelliteStatuses = make(map[string]*piraeusv1alpha1.SatelliteStatus)
-	}
```

Comment:
I think we should initialize all lists in the CR structs when we create/get them. At the moment, if there is no controller pod (e.g. because the priority class doesn't exist), then you end up with
```
status:
  ControllerStatus:
    nodeName: ""
    registeredOnController: false
  SatelliteStatuses: null
  errors:
  - 'CS Reconcile: waiting until there is exactly one controller pod, found 0 pods
    instead'
```

We want to avoid `null` for lists in the serialized representation, so we should use empty slices instead of `nil`. This agrees with https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices